### PR TITLE
Upgrade Security Scanner 

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description
Upgrade the security scanner to run against all images, not just the latest patch for each minor version.

# Context
This dependency upgrade includes a couple of enhancements, each of which are outlined below

### Scan Changes
We now run scans against all images in our LTS branches, as opposed to just on the latest patch version. This was changed in the scanner utility code and brings the behavior back to its original state. 

The go-utils PR linked above has context on why this change was introduced.

### Min Version Increase
We use to use 1.7 as the minimum version. However, since 1.8 is our true maintained min version, I increase the threshold to not scan images that we aren't updating

### Image Creation
We use to create images for the latest patch release in Solo-Projects and publish to the security tab in Gloo for all versions.

This has been changed to create a github issue for the latest patch in both solo-projects and gloo.

### Local Testing
```
IMAGE_REPO=quay.io/solo-io SCAN_DIR=_output/scans VERSION_CONSTRAINT="> v1.7.4, < v1.7.7" go run generate_docs.go run-security-scan -r glooe
```

This command will generate the security scan for 1.7.5 and 1.7.6, write those outputs to the _output/scans dir, and only 1.7.6 will result in a github issue being modified

The output logs from this command are:
```text
 IMAGE_REPO=quay.io/solo-io SCAN_DIR=_output/scans VERSION_CONSTRAINT="> v1.7.4, < v1.7.7" go run generate_docs.go run-security-scan -r glooe
{"level":"debug","ts":"2022-07-11T12:14:45.448-0400","caller":"cmd/generate_docs.go:313","msg":"Scanner will write results to directory: _output/scans"}
{"level":"debug","ts":"2022-07-11T12:14:45.449-0400","caller":"securityscanutils/securityscan.go:156","msg":"Processing user defined configuration for repository (solo-io, solo-projects)"}
{"level":"debug","ts":"2022-07-11T12:14:45.449-0400","caller":"securityscanutils/securityscan.go:163","msg":"Scanning github repo for releases that match version constraint: >v1.7.4 <v1.7.7"}
{"level":"debug","ts":"2022-07-11T12:14:50.436-0400","caller":"securityscanutils/securityscan.go:174","msg":"Number of github releases to scan: 2"}
{"level":"debug","ts":"2022-07-11T12:14:50.436-0400","caller":"securityscanutils/securityscan.go:193","msg":"GithubIssueWriter configured with Predicate: &{releasesByTagName:map[v1.7.6:0xc000202320]}"}
{"level":"debug","ts":"2022-07-11T12:14:50.436-0400","caller":"securityscanutils/securityscan.go:195","msg":"Completed processing user defined configuration."}
{"level":"debug","ts":"2022-07-11T12:20:35.468-0400","caller":"securityscanutils/github_writer.go:103","msg":"GithubIssueWriter attempting to create or update issue: Security Alert: 1.7.6"}
{"level":"debug","ts":"2022-07-11T12:20:41.069-0400","caller":"securityscanutils/securityscan.go:145","msg":"Completed running markdown scan for release v1.7.6 after 5m50.63868079s"}
{"level":"debug","ts":"2022-07-11T12:26:19.675-0400","caller":"securityscanutils/github_writer.go:87","msg":"GithubIssueWriter skipping release v1.7.5"}
{"level":"debug","ts":"2022-07-11T12:26:19.675-0400","caller":"securityscanutils/securityscan.go:145","msg":"Completed running markdown scan for release v1.7.5 after 5m38.596825922s"}
```

These logs indicate that both versions were scanned, and only 1 created/updated a github issue.

We can also see that https://github.com/solo-io/solo-projects/issues/2585 was recently edited (https://github.com/solo-io/solo-projects/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)

## Logging Fix (https://github.com/solo-io/go-utils/pull/487)
Ensure that gloo components expose a logging endpoint that can be used to dynamically change the log level.

In some of our Kube2e tests, I added a check that we can query the logging endpoint. This would have failed previously and demonstrates that we initialize the logging server with a non-nil handler.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/cmd/generate_docs.go [hunk 1]
```diff
@@ -292,55 +293,91 @@ func generateSecurityScanMd(opts *options) error {
...
+		logger.Fatalf("No version constraint defined")
+	}
+
+	outputDir := getScannerOutputDir()
+	logger.Debugf("Scanner will write results to directory: %s", outputDir)
+
+	var securityScanRepos []*securityscanutils.SecurityScanRepo
+
+	// Scan the gloo repository if either: the gloo repo is defined or no repo is defined
+	if targetRepo == "" || targetRepo == glooDocGen {
```
Comment (nfuden) [inline docs/cmd/generate_docs.go:316]: It feels weird that target repo is a string that we then check if its in what is essentially an enum without checking that we know of and approve the values of the enum.
If targetrepo is passed in as "myspecialrepo" then we scan no repos which is slightly odd

Comment (sam-heilbron) [reply docs/cmd/generate_docs.go:316]: Yeah I noticed that we could execute the command with `-r [REPO NAME]` but would always execute commands for both. In our make target we pass no value for `-r` and so I attempted to keep the functionality. 

I'm happy to require it to be explicit though

Comment (sam-heilbron) [reply docs/cmd/generate_docs.go]: I opted to make it explicit in 292c451787a87e86adbec39c20e36e240ae24a09

File: .github/workflows/README.md [hunk 2]
```diff
@@ -27,6 +27,11 @@ This Github Action will not run by default on a Draft Pull Request. If you would
 1. Mark the PR as `Ready for Review`
 1. Push an empty commit to run the jobs: `git commit --allow-empty -m "Trigger CI"`
 
+## [Trivy Vulnerability Scanning](./trivy-analysis-scheduled.yaml)
+A scheduled job which scans images released from both the Open Source and Enterprise repositories.
+
+To run the vulnerability locally, check out [the security scanner README](https://github.com/solo-io/gloo/tree/master/docs/cmd/securityscanutils)
```
Comment (elcasteel) [inline .github/workflows/README.md:33]: total nit but "to run the vulnerability check locally"

File: changelog/v1.12.0-beta31/upgrade-security-scanner.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+changelog:
+  - type: DEPENDENCY_BUMP
+    dependencyOwner: solo-io
+    dependencyRepo: go-utils
+    dependencyTag: v0.21.27
+    description: >
```
Comment (elcasteel) [inline changelog/v1.12.0-beta31/upgrade-security-scanner.yaml:6]: Another nit but I think these descriptions are swapped halfway through the first sentence.

File: docs/cmd/generate_docs.go [hunk 4]
```diff
@@ -351,7 +390,7 @@ func generateSecurityScanGloo(ctx context.Context) error {
 	if useCachedReleases() {
```
Comment (elcasteel) [inline docs/cmd/generate_docs.go:390]: what is this option for/ is it used?

File: changelog/v1.12.0-beta31/upgrade-security-scanner.yaml (hunk 3 above)
Comment (sam-heilbron) [reply changelog/v1.12.0-beta31/upgrade-security-scanner.yaml:6]: Yeah i could clear it up, though I don't think it's swapped? I meant to say that we now scan all images, as opposed to the old behavior where we would just scan the latest image for each LTS branch.

File: docs/cmd/generate_docs.go (hunk 4 above)
Comment (sam-heilbron) [reply docs/cmd/generate_docs.go:390]: This was  a relic of the existing code. I believe it's intended to reduce the number of Github API calls, but only querying the releases once, and then caching them. Previously, I think we would query the releases API a number oftimes (once to scan the images, one to publish the scans and then duplicate that for the enterprise repo)

File: changelog/v1.12.0-beta31/upgrade-security-scanner.yaml (hunk 3 above)
Comment (elcasteel) [reply changelog/v1.12.0-beta31/upgrade-security-scanner.yaml:6]: The change to the scan is implemented here in gloo and the change to the /logging endpoint is what we're getting from go-utils.

**Review by elcasteel — APPROVED** [review]

**Pull request merged**